spec.html

```diff
@@ -20989,13 +20989,24 @@
             </tr>
             <tr>
               <td>
-                [[Evaluated]]
+                [[Status]]
               </td>
               <td>
-                Boolean
+                String
+              </td>
+              <td>
+                Initially `"uninstantiated"`. Can become `"instantiated"`, `"instantiating"`, `"evaluated"`, or `"errored"` as the module progresses throughout its lifecycle.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[ErrorCompletion]]
+              </td>
+              <td>
+                An abrupt completion
               </td>
               <td>
-                Initially *false*, *true* if evaluation of this module has started. Remains *true* when evaluation completes, even if it is an abrupt completion.
+                A completion record representing exception that occurred during instantiation or evaluation; meaningful only if [[Status]] is `"errored"`.
```

Comment:
I don't think you can distinguish undefined and NormalCompletion(undefined).

I don't think there's a way to mash this into one field unless we make it multi-typed, e.g. "uninstantiated", "instantiating", "instantiated", "evaluated", or an abrupt completion. Which seems worse to me.

This kind of thing is a reasonably-common pattern in stateful systems. If you end up having too many such fields you'll want to refactor to something polymorphic, but just one seems fine.